src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -105,7 +105,14 @@ public void ActivityStarted(Activity activity)
         {
             if (this.IsEnabled(EventLevel.Verbose, EventKeywords.All))
             {
-                this.ActivityStarted(activity.OperationName, activity.Id);
+                // Accessing activity.Id here will cause the Id to be initialized
+                // before the sampler runs in case where the activity is created using legacy way
+                // i.e. new Activity("Operation name"). This will result in Id not reflecting the
+                // correct sampling flags
+                // https://github.com/dotnet/runtime/issues/61857
+                var activityId = string.Concat("00-", activity.TraceId.ToHexString(), "-", activity.SpanId.ToHexString());
```

Comment:
The Id has a fixed-length so if you want to make this more performant (on netstandard2.1+) you could use: [string.Create](https://docs.microsoft.com/en-us/dotnet/api/system.string.create?view=net-6.0)

Reply:
@CodeBlanch - we do not target netstandard 2.1 yet on sdk. 
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/OpenTelemetry.csproj

Reply:
(simply sharing)
I found this article very helpful for me to learn why String.Create is more performant in this scenario.
https://www.stevejgordon.co.uk/creating-strings-with-no-allocation-overhead-using-string-create-csharp

Reply:
@vishweshbankwar Ah shoot sorry. This only executes when user is listening to the event right? Probably not worth adding the target just for this case.

Reply:
that is right. we have a check.

Reply:
I think this is good enough (and very readable) https://github.com/dotnet/runtime/blob/f24b363c99df37dce5bc974eb879188110801d36/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L198.